### ISSUE TITLE
Feature request: user resizable panels in webui

### ISSUE THREAD

**Issue by mark-epstein** [bitfocus/companion#2781]
### Is this a feature relevant to companion itself, and not a module?

- [X] I believe this to be a feature for companion, not a module

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the feature

![image](https://github.com/bitfocus/companion/assets/80222240/49b06a84-7b77-4c29-8186-e7759cbb9d8c)
I don't know enough about web browser capabilities to even know if this is possible. But if it is, it would be very cool to have and I ask only in the microscopic hope that someone just didn't think to do it.

Consider the web page above. It would be ideal if I could grab a handle along the green highlight, and drag the dividing line to the right or left.

### Usecases

Essentially, I cannot always work with a full-screen browser and I'd like to maximize the space I give to whichever side I think is more important for what I'm doing at the time.

**Comment by LuuisElektronik** [comment]
I have the same "issue" as well. After updating, the Streamdeck XL no longer fits on the page with no option to resize

**Comment by AllanPill** [comment]
+1 for this one! 
I recently started a Streamdeck setup where trigger names get quite long to be clear and easy to find. 
That means trigger function buttons (copy, enable toggle etc.) are hidden and I need to scroll to get to them. 
When doing so, I no longer see the full name of the trigger. Found myself trying to resize the area many times, only to remember it can't be done.

**Comment by bob454522** [comment]
Im having this same issue-  

but am curious-  i just updated from 3.1.0 beta to 3.22 stable:  I could swear that prior to this update the companion web GUI would resize these buttons (ie i could use the web gui to edit buttons without having to make the browser window very large / very-wide) 
(I did read that there were some changes to the fonts and rendering and perhaps this is why)

tks

**Comment by Julusian** [comment]
@bob454522 yes, in 3.1 and earlier the ui would shrink the buttons to fix. This was possible to do automatically because of the fixed number of buttons that this needed to handle.

------

I have another solution to this which is to add a zoom control to the grid, so that the size each button is shown at can be adjusted.
But I am stuck not knowing where to put this control in the ui, so the change has been sat waiting for inspiration or a suggestion for a couple of months.
https://github.com/bitfocus/companion/pull/2745 If anyone has an idea on how to fix the layout of this section so that it looks reasonable, please post a comment there so that this zoom feature can be finished

**Comment by AJolly** [comment]
I'd love if it always lets me see all the buttons by default.

**Comment by AJolly** [comment]
<img width="2030" height="1086" alt="Image" src="https://github.com/user-attachments/assets/946f2735-7bf7-4eb1-a4c1-7a46d5a03cf5" />

A few thoughts - we can reduce the padding separating sections 1 or 3, and certainly reduce the padding for the numbers (2).  Also maybe we resize the page section to be a little smaller?
